### PR TITLE
Op 1255 fix zoom to device

### PR DESCRIPTION
**Pull request by jasperswart**
in this PR:
- fixed logic for all device types
- removed zoom in cameras because fitBounds already zooms to the best zoom level
- fixed correct device type heading in detail: devices or cameras

### CODE REVIEW

File: src/containers/MapContainer/PointClusterLayer.tsx [hunk 1]
```diff
@@ -57,14 +57,14 @@ const PointClusterLayer: React.FC<PointClusterLayerProps> = ({ onItemSelected })
       const isMarker = !f.getBounds;
       if (!f.feature) return;
       const featureId = String(isMarker ? f.feature.id : f.feature.properties.id);
-      const zoom = isMarker ? 16 : 14;
-      if (featureId === id && source === f.feature.contact) {
+      if (featureId === id) {
```
Comment (aditudorache) [inline src/containers/MapContainer/PointClusterLayer.tsx]: This won't work because there are duplicated id's in the layer, het match should be done on contact as well. The contact is set in `services/layer-aggregator/layersConfig.js`. 
As an improvement I would set the feature.properties.id to feature.id in the layersConfig.js transformers for uniformity

File: src/containers/MapContainer/PointClusterLayer.tsx [hunk 2]
```diff
@@ -50,21 +53,40 @@ const PointClusterLayer: React.FC<PointClusterLayerProps> = ({ onItemSelected })
...
+          if (fId === id && source === marker.feature.contact) {
+console.log('FOUND CLUSTER id', id);
+            onMarkerSelected(mapInstance, marker, onItemSelected, true);
+            return;
+          }
+        });
+      }
+
+      if (featureId === id && source === f.feature.contact) {
+console.log('FOUND not cluster id', id);
```
Comment (aditudorache) [inline src/containers/MapContainer/PointClusterLayer.tsx]: forgot to remove?

File: src/containers/MapContainer/PointClusterLayer.tsx [hunk 3]
```diff
@@ -50,21 +53,40 @@ const PointClusterLayer: React.FC<PointClusterLayerProps> = ({ onItemSelected })
...
       const featureId = String(isMarker ? f.feature.id : f.feature.properties.id);
-      if (featureId === id) {
-        const element = isMarker ? f._icon : f.getElement();
-        const bounds = isMarker ? [f.getLatLng(), f.getLatLng()] : f.getBounds();
-        mapInstance.fitBounds(bounds);
-        if (isMarker) {
-          mapInstance.setZoom(16);
-        }
-        onItemSelected(isMarker ? 'devices' : 'cameras', f.feature, element);
+      if (isMarker) {
```
Comment (aditudorache) [inline src/containers/MapContainer/PointClusterLayer.tsx]: Does this cover the case when the marker is not inside a cluster on the default zoom level?

File: src/containers/MapContainer/PointClusterLayer.tsx [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
 import { useMapInstance } from '@datapunt/react-maps';
 import { useDispatch, useSelector } from 'react-redux';
-import { MarkerClusterGroupOptions } from 'leaflet';
+import { MarkerClusterGroupOptions, marker } from 'leaflet';
 import layersReader from 'services/layer-aggregator/layersReader';
 import queryStringParser from '../../shared/services/auth/services/query-string-parser';
 import LAYERS_CONFIG, { getPointOptions } from '../../services/layer-aggregator/layersConfig';
 import { addLayerDataActionCreator, removeLayerDataActionCreator } from './MapContainerDucks';
 import MarkersCluster, { MarkerClusterData } from '../../components/MarkerCluster/MarkersCluster';
+import { isMemberExpression } from '@babel/types';
```
Comment (aditudorache) [inline src/containers/MapContainer/PointClusterLayer.tsx]: why is this needed?

File: src/containers/MapContainer/PointClusterLayer.tsx [hunk 5]
```diff
@@ -53,12 +53,12 @@ const PointClusterLayer: React.FC<PointClusterLayerProps> = ({ onItemSelected })
     state?.map?.layers.filter((l:any) => layerNames.current.includes(l.name) && state?.map?.legend[l.name]),
   );
 
-  const onMarkerSelected = (map: any, layer: any, onMarkerSelected: Function, isMarker: boolean) => {
+  const hanleMarkerSelected = (map: any, layer: any, onMarkerSelected: Function, isMarker: boolean) => {
```
Comment (aditudorache) [inline src/containers/MapContainer/PointClusterLayer.tsx]: should be `handleMarkerSelected`?

File: src/containers/MapContainer/PointClusterLayer.tsx [hunk 6]
```diff
@@ -53,12 +53,12 @@ const PointClusterLayer: React.FC<PointClusterLayerProps> = ({ onItemSelected })
...
     const element = isMarker ? layer._icon : layer.getElement();
     const bounds = isMarker ? [layer.getLatLng(), layer.getLatLng()] : layer.getBounds();
     map.fitBounds(bounds);
     if (isMarker) {
-      map.setZoom(16);
+       map.setZoom(16);
```
Comment (aditudorache) [inline src/containers/MapContainer/PointClusterLayer.tsx]: extra space here?

File: src/containers/MapContainer/PointClusterLayer.tsx (hunk 5 above)
Comment (jasperswart) [reply src/containers/MapContainer/PointClusterLayer.tsx]: gedaan

File: src/containers/MapContainer/PointClusterLayer.tsx (hunk 3 above)
Comment (jasperswart) [reply src/containers/MapContainer/PointClusterLayer.tsx]: clustering only happens with markers and nog areas

File: src/containers/MapContainer/PointClusterLayer.tsx (hunk 1 above)
Comment (jasperswart) [reply src/containers/MapContainer/PointClusterLayer.tsx]: it is back in there

File: src/containers/MapContainer/PointClusterLayer.tsx (hunk 2 above)
Comment (jasperswart) [reply src/containers/MapContainer/PointClusterLayer.tsx]: sorry

File: src/containers/MapContainer/PointClusterLayer.tsx (hunk 4 above)
Comment (jasperswart) [reply src/containers/MapContainer/PointClusterLayer.tsx]: incorrect import sorry

File: src/containers/MapContainer/PointClusterLayer.tsx (hunk 6 above)
Comment (jasperswart) [reply src/containers/MapContainer/PointClusterLayer.tsx]: fixed that

File: src/containers/MapContainer/PointClusterLayer.tsx (hunk 3 above)
Comment (aditudorache) [reply src/containers/MapContainer/PointClusterLayer.tsx]: I got that but if the marker is not in a cluster, this branch won't highlight it. Pease check 
- http://localhost:3001/?id=100&category=Camera&source=bfa 
- http://localhost:3001/?id=706&category=Baken&source=iothings
for highlight

File: src/containers/MapContainer/PointClusterLayer.tsx (hunk 5 above)
Comment (aditudorache) [reply src/containers/MapContainer/PointClusterLayer.tsx]: I see no change here

Comment (jasperswart) [reply src/containers/MapContainer/PointClusterLayer.tsx]: the name is now hanleMarkerSelected

Comment (aditudorache) [reply src/containers/MapContainer/PointClusterLayer.tsx]: I thought **handle**MarkerSelected

File: src/containers/MapContainer/PointClusterLayer.tsx (hunk 3 above)
Comment (jasperswart) [reply src/containers/MapContainer/PointClusterLayer.tsx]: I fixed that issue with highlighting markers and areas
